pilot/pkg/config/kube/gateway/backend_policies.go

```diff
@@ -0,0 +1,423 @@
+	Source       TypedNamedspacedName
+	TargetIndex  int
+	Target       TypedNamedspacedName
+	TLS          *networking.ClientTLSSettings
+	LoadBalancer *networking.LoadBalancerSettings
+	CreationTime time.Time
+}
+
+func (b BackendPolicy) ResourceName() string {
+	return b.Source.String() + "/" + fmt.Sprint(b.TargetIndex)
+}
+
+func (b BackendPolicy) Equals(other BackendPolicy) bool {
+	return b.Source == other.Source &&
+		protoconv.Equals(b.TLS, other.TLS) &&
+		protoconv.Equals(b.LoadBalancer, other.LoadBalancer)
+}
+
+// DestinationRuleCollection returns a collection of DestinationRule objects. These are built from a few different
+// policy types that are merged together.
+func DestinationRuleCollection(
+	trafficPolicies krt.Collection[*gatewayx.XBackendTrafficPolicy],
+	tlsPolicies krt.Collection[*gatewayalpha3.BackendTLSPolicy],
+	references *ReferenceSet,
+	domainSuffix string,
+	c *Controller,
+	opts krt.OptionsBuilder,
+) krt.Collection[*config.Config] {
+	trafficPolicyStatus, backendTrafficPolicies := BackendTrafficPolicyCollection(trafficPolicies, references, opts)
+	registerStatus(c, trafficPolicyStatus)
+
+	tlsPolicyStatus, backendTLSPolicies := BackendTLSPolicyCollection(tlsPolicies, references, opts)
+	registerStatus(c, tlsPolicyStatus)
+
+	// We need to merge these by hostname into a single DR
+	allPolicies := krt.JoinCollection([]krt.Collection[BackendPolicy]{backendTrafficPolicies, backendTLSPolicies})
+	byTarget := krt.NewIndex(allPolicies, func(o BackendPolicy) []TypedNamedspacedName {
+		return []TypedNamedspacedName{o.Target}
+	})
+	indexOpts := append(opts.WithName("BackendPolicyByTarget"), krt.WithIndexCollectionFromString(func(s string) TypedNamedspacedName {
+		parts := strings.Split(s, "/")
+		if len(parts) != 3 {
+			panic("invalid TypedNamedspacedName: " + s)
+		}
+		return TypedNamedspacedName{
+			NamespacedName: types.NamespacedName{
+				Namespace: parts[1],
+				Name:      parts[2],
+			},
+			Kind: kind.FromString(parts[0]),
+		}
+	}))
+	merged := krt.NewCollection(
+		byTarget.AsCollection(indexOpts...),
+		func(ctx krt.HandlerContext, i krt.IndexObject[TypedNamedspacedName, BackendPolicy]) **config.Config {
+			svc := i.Key
+			// Sort so we can pick the oldest, which will win
+			pols := slices.SortFunc(i.Objects, func(a, b BackendPolicy) int {
+				if r := a.CreationTime.Compare(b.CreationTime); r != 0 {
+					return r
+				}
+				if r := cmp.Compare(a.Source.Namespace, b.Source.Namespace); r != 0 {
+					return r
+				}
+				return cmp.Compare(a.Source.Name, b.Source.Name)
+			})
+			_ = pols
+			tlsSet := false
+			lbSet := false
+			spec := &networking.DestinationRule{
+				Host:          fmt.Sprintf("%s.%s.svc.%v", svc.Name, svc.Namespace, domainSuffix),
+				TrafficPolicy: &networking.TrafficPolicy{},
+			}
+			parents := make([]string, 0, len(pols))
+			for _, pol := range pols {
+				parents = append(parents, fmt.Sprintf("%s/%s.%s", pol.Source.Kind, pol.Source.Namespace, pol.Source.Name))
```

Comment:
Should we append to parents if we may skip the policy later?

Reply:
thanks, lost track fo this one. fixed